typings/index.d.ts

```diff
@@ -1982,14 +1982,27 @@ declare module 'discord.js' {
     constructor(channel: TextChannel | DMChannel, iterable?: Iterable<any>);
     public channel: TextBasedChannelFields;
     public cache: Collection<Snowflake, Message>;
+    public crosspost(message: MessageResolvable): Promise<void>;
```

Comment:
This following suggestions rely on the [`APIMessage`](https://github.com/discordjs/discord.js/pull/4876/files#diff-494415c3af35245ecae73f5d42d8b331R2194-R2281) interface over on #4876.
```suggestion
    public crosspost(message: MessageResolvable): Promise<Message | APIRawMessage>;
```

Reply:
If this is done, I think it'll be suitable to add an overload, where passing a `Message` instance makes it always return a `Promise<Message>`.